Document air_quality triggers & conditions (#44955)
home-assistant/home-assistant.github.io: source/_conditions/air_quality.is_co2_value.markdown

---
title: "Carbon dioxide value"
condition: air_quality.is_co2_value
domain: air_quality
description: "Tests the carbon dioxide level of one or more entities."
related_conditions:
  - air_quality.is_voc_value
---

> The **Carbon dioxide value** condition passes when a carbon dioxide (CO2) sensor's reading meets a specific level. A stuffy meeting room, a crowded living room on movie night, or a bedroom with the door closed overnight all push CO2 levels higher than you would expect. This condition lets your automation act only when CO2 is genuinely elevated, so the ventilation fan kicks in when it is truly needed and stays off when the air is fine.

Reviewer: "kicks in" -> "starts"

```suggestion
The **Carbon dioxide value** condition passes when a carbon dioxide (CO2) sensor's reading meets a specific level. A stuffy meeting room, a crowded living room on movie night, or a bedroom with the door closed overnight all push CO2 levels higher than you would expect. This condition lets your automation act only when CO2 is genuinely elevated, so the ventilation fan starts when it is truly needed and stays off when the air is fine.
```

{% include integrations/labs_entity_triggers_note.md %}

{% include conditions/ui_header.md %}

> To use this condition in an automation:

Reviewer 2: ```suggestion
## Using this condition in an automation:
```
I would use a task title here, like we do for other tasks

    threshold: 1000
    behavior: any
{% endexample %}

This passes when the office CO2 sensor reads at or above 1000 ppm.

### Options in YAML

YAML sometimes provides additional options for more complex use cases that are not available through the UI.

{% options_yaml %}
threshold:
  description: >
    The carbon dioxide level the sensor has to meet or exceed for the condition to pass. Accepts a number or a reference to an `input_number`, `number`, or `sensor` entity.
  required: true
  type: any
behavior:
  description: >
    When multiple sensors are targeted, controls how results combine. Accepts `all` or `any`.
  required: true
  type: string
  default: any
{% endoptions_yaml %}

{% include conditions/targets.md %}

{% include conditions/behavior.md %}

## Good to know

- Sensors that are unavailable (`unavailable`) or have an unknown state (`unknown`) are skipped for **Any** and fail for **All**.
- Outdoor CO2 levels are typically around 420 ppm. Indoor levels above 1000 ppm often suggest the room needs better ventilation.

{% include conditions/try_it.md %}

{% include conditions/more_examples.md %}

### Automation: turn on ventilation at bedtime only if CO2 is elevated

> After an evening spent in the living room with the doors closed, CO2 levels are sometimes higher than you would expect by the time you head to bed. This automation triggers at your usual bedtime and checks the bedroom CO2 reading. If the level is at or above 1000 ppm, the ventilation fan turns on so you sleep with fresh air. On evenings when the room already has good airflow, the fan stays off.

Reviewer: Make first sentence more grammatically correct:

```suggestion
After you spend an evening in the living room with the doors closed, CO2 levels are sometimes higher than you would expect by the time you head to bed. This automation triggers at your usual bedtime and checks the bedroom CO2 reading. If the level is at or above 1000 ppm, the ventilation fan turns on so you sleep with fresh air. On evenings when the room already has good airflow, the fan stays off.
```
